packages/yourloops/pages/profile/profile.tsx

```diff
@@ -240,98 +197,79 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
-  const createHandleSelectChange = <T extends Units | LanguageCodes | HcpProfession>(setState: SetState<T>): HandleChange<SelectChangeEvent> => (event) => setState(event.target.value as T);
-
   const handleSwitchRoleOpen = () => {
     setSwitchRoleOpen(true);
     metrics.send("switch_account", "display_switch_preferences");
   };
-  const handleSwitchRoleCancel = () => setSwitchRoleOpen(false);
 
-  const errors: Errors = React.useMemo(
-    () => ({
-      firstName: _.isEmpty(firstName),
-      lastName: _.isEmpty(lastName),
-      hcpProfession: role === UserRoles.hcp && hcpProfession === HcpProfession.empty,
-      currentPassword: password.length > 0 && currentPassword.length < appConfig.PWD_MIN_LENGTH,
-      password: passwordCheckResults.onError,
-      passwordConfirmation: passwordConfirmation !== password,
-      birthDate: role === UserRoles.patient && !REGEX_BIRTHDATE.test(birthDate),
-    }),
-    [firstName, lastName, hcpProfession, password, currentPassword.length, passwordCheckResults.onError, passwordConfirmation, role, birthDate]
-  );
+  const handleSwitchRoleCancel = () => setSwitchRoleOpen(false);
 
-  const isAnyError = React.useMemo(() => _.some(errors), [errors]);
-  const canSave = (preferencesChanged || profileChanged || settingsChanged || passwordChanged) && !isAnyError;
+  const preferencesChanged = !_.isEqual(user.preferences, getUpdatedPreferences());
+  const profileChanged = !_.isEqual(user.profile, getUpdatedProfile());
+  const settingsChanged = !_.isEqual(user.settings, getUpdatedSettings());
+  const passwordChanged = password !== "" || passwordConfirmation !== "";
+  const isAnyError = useMemo(() => _.some(errors), [errors]);
+  const canSave = (preferencesChanged || profileChanged || settingsChanged || passwordChanged) && !isAnyError && !saving;
 
   const onSave = async (): Promise<void> => {
     let preferences: Preferences | null = null;
     let profile: Profile | null = null;
     let settings: Settings | null = null;
-    let updateFailed = false;
-    /** Set to true if we need to update the user only (no change needed for the password) */
-    let updated = false;
+
+    const updatedUser = new User(user);
 
     try {
-      if (preferencesChanged) {
-        preferences = await updatePreferences(getUpdatedPreferences(), false);
-        updated = true;
-      }
+      setSaving(true);
+
       if (profileChanged) {
         profile = await updateProfile(getUpdatedProfile(), false);
-        updated = true;
+        updatedUser.profile = profile;
       }
+
       if (settingsChanged) {
         settings = await updateSettings(getUpdatedSettings(), false);
-        updated = true;
+        updatedUser.settings = settings;
       }
+
       if (role !== UserRoles.patient && passwordChanged) {
         await updatePassword(currentPassword, password);
+        setCurrentPassword("");
```

Comment:
Your code seems to be working without these three lines :thinking:

Reply:
you're right it works ;)